test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -2094,9 +2123,24 @@ static int traffic_mngr_suite_init(void)
 		       iface_name[0], iface_name[1]);
 	}
 
-	if (open_pktios() != 0)
+	pktios[0] = ODP_PKTIO_INVALID;
+	pktios[1] = ODP_PKTIO_INVALID;
+
+	ret = open_pktios();
+	if (ret < 0)
 		return -1;
 
+	/* Positive return indicates, that pktio open failed with out mode as TM
+	 * but succeeded with direct mode.
+	 */
+	if (ret > 0)
+		goto skip_tests;
+
+	return 0;
+skip_tests:
+	/* Mark all tests as inactive under this suite */
+	odp_cunit_set_inactive();
```

Comment:
I am not sure we really need this new API in the common test framework. There are at least two ways this code could effectively disable the test without having the new API:
1) Do the pktio open etc before suite init so that suite init and test case registration is not even done if the pktio is not suitable
2) Use ODP_TEST_INFO_CONDITIONAL and the check function to disable every test at registration time if the init phase has failed.

Reply:
Method '1' is not sufficient as in changes in #1237 for every suite I validate whether platform supports enough levels, modes etc,  If I do it in main before cunit registration, then it will be not clean I guess.

Method '2' can be done but just that every test case will become ODP_TEST_INFO_CONDITIONAL() though the it is a global inactive decision.

Reply:
Ok, let's have this as proposed.